modules/agar/src/main/ts/ephox/agar/find/UiSearcher.ts

```diff
@@ -73,8 +73,25 @@ const findIn = <T extends Element>(container: SugarElement<Node>, selector: stri
 const findAllIn = <T extends Element>(container: SugarElement<Node>, selector: string): Array<SugarElement<T>> =>
   selectAll(container, selector);
 
+const findTargetByLabel = <T extends Element>(container: SugarElement<Node>, labelText: string): Result<SugarElement<T>, TestLabel> => {
+  const label = Arr.find(selectAll(container, 'label'), (e) => e.dom.textContent === labelText);
```

Comment:
Since we are using sugar elements, I'd rather you use `TextContent.get` like so:
```suggestion
  const label = Arr.find(selectAll(container, 'label'), (e) => TextContent.get(e) === labelText);
```
Of course you need to import `TextContent`

Reply:
✅